src/include/duckdb/common/types/string_type.hpp

```diff
@@ -70,6 +71,41 @@ struct string_t {
 	    : string_t(value.c_str(), UnsafeNumericCast<uint32_t>(value.size())) {
 	}
 
+	// Allocate using an arena allocator
+	string_t(const char *data, const uint32_t len, ArenaAllocator &arena) {
```

Comment:
Nit: Can we pass the `ArenaAllocator` as the first argument, iirc that's more in line with other code. :)